# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerEventTarget.cpp
# Commit: 32515f9b41c7
# Full Hash: 32515f9b41c77d6ccd5831eed28953037c4fb319
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P2 remove nsICancelableRunnable inheriting form WorkerRunnable. r=asuth
#   
#   This patch is only remove the inheriting, but keeping Cancel() implmentation.
#   The Cancel() is needed in some special cases, such as Worker enters Canceling when WorkerScope is not created.
#   
# ==============================================================================

diff -r 910dabb8545d -r 32515f9b41c7 dom/workers/WorkerEventTarget.cpp
--- a/dom/workers/WorkerEventTarget.cpp	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/workers/WorkerEventTarget.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -58,16 +58,14 @@
     // If the inner runnable is not cancellable, then just do the normal
     // WorkerControlRunnable thing.  This will end up calling Run().
     if (!cr) {
-      WorkerControlRunnable::Cancel();
-      return NS_OK;
+      return Run();
     }
 
     // Otherwise call the inner runnable's Cancel() and treat this like
     // a WorkerRunnable cancel.  We can't call WorkerControlRunnable::Cancel()
     // in this case since that would result in both Run() and the inner
     // Cancel() being called.
-    Unused << cr->Cancel();
-    return WorkerRunnable::Cancel();
+    return cr->Cancel();
   }
 };
 
@@ -115,6 +113,9 @@
   }
 
   if (mBehavior == Behavior::Hybrid) {
+    LOGV(("WorkerEventTarget::Dispatch [%p] Dispatch as normal runnable(%p)",
+          this, runnable.get()));
+
     RefPtr<WorkerRunnable> r =
         mWorkerPrivate->MaybeWrapAsWorkerRunnable(runnable.forget());
     if (r->Dispatch()) {